Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.h

```diff
@@ -82,8 +96,25 @@ namespace AzToolsFramework
             return m_jsonBackend;
         }
 
+        QModelIndex GetIndexFromNode(DPEModelNode* const theNode) const;
+
+        void SetMaxColumns(int newMax)
+        {
+            m_maxColumns = newMax;
+        }
```

Comment:
Shouldn't this protect against negative column values?

Reply:
If your point is that it could be unsigned, I'm keeping it the same type that Qt uses everywhere else to describe rows and columns in model indices, which is int:
https://doc.qt.io/qt-5/qmodelindex.html

If your point is that I should validate all input to my API, why not pick on any of the other functions here? Calling SetAdapter with a random pointer would definitely mess things up too.